### PR TITLE
v20240627 notes

### PR DESCRIPTION
**Pull request by shashankbarsin**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,46 @@
...
+### Announcements
+
+* Starting 1.30 Kubernetes version and 1.27 LTS versions, beta APIs will be disabled by default, when you upgrade to [them](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#before-you-begin). There will be an option provided to explicitly enable beta APIs closer to the 1.30 release.
+* 1.30 is the next LTS version after 1.27. Customers will have at least 6 months of migration time available between 1.27 LTS to 1.30 LTS i.e before the 1.27 LTS goes End of Life by July 2025. Refer [long term support](https://learn.microsoft.com//azure/aks/long-term-support) for more information.
+
+### Release Notes
+
+* Features:
+  * AKS patch versions 1.29.5, 1.28.10, 1.27.14 are now available. Refer to [version support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#kubernetes-version-support-policy) and [upgrading a cluster](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli) for more information.
+
```
Comment (allyford) [inline CHANGELOG.md:16]: Preview feature:
[Windows Server Annual Channel for Containers](https://techcommunity.microsoft.com/t5/windows-server-news-and-best/windows-server-annual-channel-for-containers/ba-p/3866248) is now in public preview on Azure Kubernetes Service (AKS). To learn more, visit: https://aka.ms/aks/windowsannualchannel.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,10 +1,49 @@
 # Azure Kubernetes Service Changelog
 
+## Release 2024-06-27
+
+Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled as v20240627.
+
+### Announcements
+
+* Starting 1.30 Kubernetes version and 1.27 LTS versions, beta APIs will be disabled by default, when you upgrade to [them](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#before-you-begin). There will be an option provided to explicitly enable beta APIs closer to the 1.30 release.
+* 1.30 is the next LTS version after 1.27. Customers will have at least 6 months of migration time available between 1.27 LTS to 1.30 LTS i.e before the 1.27 LTS goes End of Life by July 2025. Refer [long term support](https://learn.microsoft.com//azure/aks/long-term-support) for more information.
```
Comment (pavneeta) [inline CHANGELOG.md:10]: I am not sure I follow this completely - 1.30 LTS is releasing when ? if 1.27 is going EOL in July 2025 - so dont cusotmers have an year of overlap?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,10 +1,49 @@
...
+  
+* Bug Fixes:
+  * Fixed a [bug](https://github.com/Azure/AKS/issues/4292) that previously didn't allow switching from non-LTS K8s version to [LTS](https://learn.microsoft.com/azure/aks/long-term-support) K8s version when upgrading the cluster. For example, you can now upgrade from 1.26 to 1.27 while switching to LTS.
+  * Related to the above, also fixed a bug where previously it was not possible to upgrade from an LTS K8s version to non-LTS K8s version. For example, you can now upgrade from 1.27 LTS to 1.28.
+
+* Behavior Change
+  * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/azure/aks/csi-secrets-store-driver) has been updated from 300Mi to 500Mi.
+  * Base CPU and memory for metrics-server container are set to 150M and 100Mi respectively on clusters with K8s version >= 1.30.0. More information on metrics server scaling can be found [here](https://learn.microsoft.com/azure/aks/use-metrics-server-vertical-pod-autoscaler).
```
Comment (pavneeta) [inline CHANGELOG.md:26]: what was the previous behavior? I am assuming there was no limit?

File: CHANGELOG.md (hunk 2 above)
Comment (shashankbarsin) [reply CHANGELOG.md:10]: 1.30 isn't GA yet. Tagged you on an internal thread about this topic for follow-up

Comment (sabbour) [reply CHANGELOG.md]: +1

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,13 +1,52 @@
...
+
+* Preview feature:
+  * [Windows Server Annual Channel for Containers](https://techcommunity.microsoft.com/t5/windows-server-news-and-best/windows-server-annual-channel-for-containers/ba-p/3866248) is now in public preview on Azure Kubernetes Service (AKS). More information can be found [here](https://aka.ms/aks/windowsannualchannel).
+  
+* Bug Fixes:
+  * Fixed a [bug](https://github.com/Azure/AKS/issues/4292) that previously didn't allow switching from non-LTS K8s version to [LTS](https://learn.microsoft.com/azure/aks/long-term-support) K8s version when upgrading the cluster. For example, you can now upgrade from 1.26 to 1.27 while switching to LTS.
+  * Related to the above, also fixed a bug where previously it was not possible to upgrade from an LTS K8s version to non-LTS K8s version. For example, you can now upgrade from 1.27 LTS to 1.28.
+
+* Behavior Change
+  * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/azure/aks/csi-secrets-store-driver) has been updated from 300Mi to 500Mi.
```
Comment (sabbour) [inline CHANGELOG.md:25]: For which Kubernetes version?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,13 +1,52 @@
...
+  * Base CPU and memory for metrics-server container are updated from 44M to 150M and 51Mi to 100Mi respectively on clusters with K8s version >= 1.30.0. More information on metrics server scaling can be found [here](https://learn.microsoft.com/azure/aks/use-metrics-server-vertical-pod-autoscaler).
+
+     
+* Component Updates:
+  * Linux Network Policy Manager has been upgraded from v1.5.23 to [v.1.5.29](https://github.com/Azure/azure-container-networking/releases/tag/v1.5.29) to address [CVE-2024-28085](https://nvd.nist.gov/vuln/detail/CVE-2024-28085).
+  * Upgraded Azure workload identity to [v1.3.0](https://github.com/Azure/azure-workload-identity/releases/tag/v1.3.0).
+  * Upgraded ip-masq-agent-v2 to [v0.1.11](https://github.com/Azure/ip-masq-agent-v2/releases/tag/v0.1.11) having fixes for [CVE-2024-2961](https://nvd.nist.gov/vuln/detail/CVE-2024-2961) and [CVE-2024-33599](https://nvd.nist.gov/vuln/detail/CVE-2024-33599).
+  * Upgraded Azure Monitor Container Insights image to [v3.1.22](https://github.com/microsoft/Docker-Provider/releases/tag/3.1.22)
```
Comment (sabbour) [inline CHANGELOG.md:33]: For which Kubernetes version?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,13 +1,52 @@
...
+  * Related to the above, also fixed a bug where previously it was not possible to upgrade from an LTS K8s version to non-LTS K8s version. For example, you can now upgrade from 1.27 LTS to 1.28.
+
+* Behavior Change
+  * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/azure/aks/csi-secrets-store-driver) has been updated from 300Mi to 500Mi.
+  * Base CPU and memory for metrics-server container are updated from 44M to 150M and 51Mi to 100Mi respectively on clusters with K8s version >= 1.30.0. More information on metrics server scaling can be found [here](https://learn.microsoft.com/azure/aks/use-metrics-server-vertical-pod-autoscaler).
+
+     
+* Component Updates:
+  * Linux Network Policy Manager has been upgraded from v1.5.23 to [v.1.5.29](https://github.com/Azure/azure-container-networking/releases/tag/v1.5.29) to address [CVE-2024-28085](https://nvd.nist.gov/vuln/detail/CVE-2024-28085).
```
Comment (sabbour) [inline CHANGELOG.md:30]: For which Kubernetes version?

File: CHANGELOG.md (hunk 4 above)
Comment (shashankbarsin) [reply CHANGELOG.md:25]: from the PR for this change, I see it being applied for all k8s versions

File: CHANGELOG.md (hunk 6 above)
Comment (shashankbarsin) [reply CHANGELOG.md:30]: this is a pre-pulled image in the vhd

File: CHANGELOG.md (hunk 5 above)
Comment (shashankbarsin) [reply CHANGELOG.md:33]: patch upgrade for all k8s versions

File: CHANGELOG.md
Comment (miwithro) [inline CHANGELOG.md:1]: @shashankbarsin we need to add the following announcement.

Starting in mid-September 2024 the Microsoft Defender for Containers Image will update to the latest image in order to migrate to the [Inspector Gadget](https://github.com/inspektor-gadget/inspektor-gadget) Collector.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,13 +1,52 @@
...
+
+### Announcements
+
+* Starting 1.30 Kubernetes version and 1.27 LTS versions, beta APIs will be disabled by default, when you upgrade to [them](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#before-you-begin). There will be an option provided to explicitly enable beta APIs closer to the 1.30 release.
+* 1.30 is the next LTS version after 1.27. Customers will have at least 6 months of migration time available between 1.27 LTS to 1.30 LTS i.e before the 1.27 LTS goes End of Life by July 2025. Refer [long term support](https://learn.microsoft.com/azure/aks/long-term-support) for more information.
+
+### Release Notes
+
+* Features:
+  * AKS patch versions 1.29.5, 1.28.10, 1.27.14 are now available. Refer to [version support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#kubernetes-version-support-policy) and [upgrading a cluster](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli) for more information.
```
Comment (kaarthis) [inline CHANGELOG.md:15]: did we deprecate any patch.. if you see anything in the PR list pls call it out too.

File: CHANGELOG.md (hunk 5 above)
Comment (kaarthis) [inline CHANGELOG.md:33]: on all the upgrades is there any impact / improvements to see type call out we can provide for customer.  Just to answe the question so what?

File: CHANGELOG.md [hunk 8]
```diff
@@ -20,7 +59,7 @@ Monitor the release status by regions at [AKS-Release-Tracker](https://releases.
 * Preview Features:
   * AKS version [1.30](https://kubernetes.io/blog/2024/04/17/kubernetes-v1-30-release/) is available in preview.
```
Comment (kaarthis) [inline CHANGELOG.md:60]: Why do we need to call this out.. The last release 15 days ago had the same line.

File: CHANGELOG.md (hunk 7 above)
Comment (shashankbarsin) [reply CHANGELOG.md:15]: I see 1.27.9, 1.28.5, and 1.29.2 in the red. @kaarthis can you follow up with the team and confirm that these are being removed? and I'll add it here

File: CHANGELOG.md (hunk 5 above)
Comment (shashankbarsin) [reply CHANGELOG.md:33]: no major changes in this specific release. have tagged the release notes for all the upgraded components for more information

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,13 +1,52 @@
...
+
+* Preview feature:
+  * [Windows Server Annual Channel for Containers](https://techcommunity.microsoft.com/t5/windows-server-news-and-best/windows-server-annual-channel-for-containers/ba-p/3866248) is now in public preview on Azure Kubernetes Service (AKS). More information can be found [here](https://aka.ms/aks/windowsannualchannel).
+  
+* Bug Fixes:
+  * Fixed a [bug](https://github.com/Azure/AKS/issues/4292) that previously didn't allow switching from non-LTS K8s version to [LTS](https://learn.microsoft.com/azure/aks/long-term-support) K8s version when upgrading the cluster. For example, you can now upgrade from 1.26 to 1.27 while switching to LTS.
+  * Related to the above, also fixed a bug where previously it was not possible to upgrade from an LTS K8s version to non-LTS K8s version. For example, you can now upgrade from 1.27 LTS to 1.28.
+
+* Behavior Change
```
Comment (robbiezhang) [inline CHANGELOG.md:24]: seems like we missed this breaking change:
https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp/pullrequest/10229606

File: CHANGELOG.md (hunk 8 above)
Comment (shashankbarsin) [reply CHANGELOG.md:60]: this line hasn't changed. you are commenting on the previous release :)

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,13 +1,52 @@
...
+Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled as v20240627.
+
+### Announcements
+
+* Starting 1.30 Kubernetes version and 1.27 LTS versions, beta APIs will be disabled by default, when you upgrade to [them](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#before-you-begin). There will be an option provided to explicitly enable beta APIs closer to the 1.30 release.
+* 1.30 is the next LTS version after 1.27. Customers will have at least 6 months of migration time available between 1.27 LTS to 1.30 LTS i.e before the 1.27 LTS goes End of Life by July 2025. Refer [long term support](https://learn.microsoft.com/azure/aks/long-term-support) for more information.
+
+### Release Notes
+
+* Features:
```
Comment (kaysieyu) [inline CHANGELOG.md:14]: @shashankbarsin can you add the following:
[AKS Cost Analysis](https://learn.microsoft.com/azure/aks/cost-analysis) views can now be accessed directly from the Cluster overview blade in Azure Portal. [Learn more](https://learn.microsoft.com/azure/cost-management-billing/costs/view-kubernetes-costs).

**Review by sabbour — APPROVED** [review]

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,13 +1,53 @@
...
+  
+* Bug Fixes:
+  * Fixed a [bug](https://github.com/Azure/AKS/issues/4292) that previously didn't allow switching from non-LTS K8s version to [LTS](https://learn.microsoft.com/azure/aks/long-term-support) K8s version when upgrading the cluster. For example, you can now upgrade from 1.26 to 1.27 while switching to LTS.
+  * Related to the above, also fixed a bug where previously it was not possible to upgrade from an LTS K8s version to non-LTS K8s version. For example, you can now upgrade from 1.27 LTS to 1.28.
+
+* Behavior Change
+  * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/azure/aks/csi-secrets-store-driver) has been updated from 300Mi to 500Mi.
+  * Base CPU and memory for metrics-server container are updated from 44M to 150M and 51Mi to 100Mi respectively on clusters with K8s version >= 1.30.0. More information on metrics server scaling can be found [here](https://learn.microsoft.com/azure/aks/use-metrics-server-vertical-pod-autoscaler).
```
Comment (palma21) [inline CHANGELOG.md:27]: What's "base"? Requests, limits, minimum from autoscale? Other?

**Review by palma21 — APPROVED** [review]

File: CHANGELOG.md (hunk 11 above)
Comment (shashankbarsin) [reply CHANGELOG.md:27]: https://learn.microsoft.com/en-us/azure/aks/use-metrics-server-vertical-pod-autoscaler - data.NannyConfiguration.baseCPU and data.NannyConfiguration.baseMemory. I agree we need to clarify what baseCPU and baseMemory mean. will follow up with team and add this clarification on the docs and merge release notes right now

**Pull request merged**